imagebuildah/stage_executor.go

```diff
@@ -923,6 +1034,27 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 				if fromErr != nil {
 					return "", nil, errors.Wrapf(fromErr, "unable to resolve argument %q", arr[1])
 				}
+				// If additional buildContext contains this
+				// give priority to that and break if additional
+				// is not an external image.
+				if additionalBuildContext, ok := s.executor.additionalBuildContexts[from]; ok {
+					if additionalBuildContext != nil {
```

Comment:
If we don't set `nil` values in the map, we don't need to check for them.

Reply:
Same as above.